File: bench/tasks/12-revenue/zig/main.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    const revenue = [_]i64{ 240, 175, 289, 150, 225, 199, 180, 178 };
    var total: i64 = 0;
    var high_sum: i64 = 0;
    var high_count: i64 = 0;
    var budget_count: i64 = 0;
    for (revenue) |r| {
        total += r;
        if (r >= 200) {
            high_count += 1;
            high_sum += r;
        } else {
            budget_count += 1;
        }
    }
    try stdout.print("Total revenue: {}\n", .{total});
    try stdout.print("Premium orders (200+): {}\n", .{high_count});
    try stdout.print("Premium revenue: {}\n", .{high_sum});
    try stdout.print("Budget orders: {}\n", .{budget_count});
}

File: bench/tasks/12-revenue/rust/main.rs
fn main() {
    let revenue = [240, 175, 289, 150, 225, 199, 180, 178];
    let total: i32 = revenue.iter().sum();
    let high_value: Vec<i32> = revenue.iter().copied().filter(|&r| r >= 200).collect();
    let high_sum: i32 = high_value.iter().sum();
    let budget_count = revenue.iter().filter(|&&r| r < 200).count();
    println!("Total revenue: {}", total);
    println!("Premium orders (200+): {}", high_value.len());
    println!("Premium revenue: {}", high_sum);
    println!("Budget orders: {}", budget_count);
}
